ydb/src/client_topic/topicwriter/writer.rs

```diff
@@ -0,0 +1,405 @@
+use tokio::time::error::Elapsed;
+use tokio::time::timeout;
+use tokio_util::sync::CancellationToken;
+use ydb_grpc::ydb_proto::topic::stream_write_message;
+use ydb_grpc::ydb_proto::topic::stream_write_message::from_client::ClientMessage;
+use ydb_grpc::ydb_proto::topic::stream_write_message::from_server::ServerMessage;
+use ydb_grpc::ydb_proto::topic::stream_write_message::init_request::Partitioning;
+use ydb_grpc::ydb_proto::topic::stream_write_message::write_request::{message_data, MessageData};
+use ydb_grpc::ydb_proto::topic::stream_write_message::{
+    FromClient, FromServer, InitRequest, WriteRequest,
+};
+
+pub enum TopicWriterState {
+    Working,
+    FinishedOk,
+    FinishedWithError(YdbError),
+}
+
+#[allow(dead_code)]
+pub struct TopicWriter {
+    pub(crate) path: String,
+    pub(crate) producer_id: Option<String>,
+    pub(crate) partition_id: i64,
+    pub(crate) session_id: String,
+    pub(crate) last_seq_num_handled: i64,
+    pub(crate) write_request_messages_chunk_size: usize,
+    pub(crate) write_request_send_messages_period: Duration,
+
+    pub(crate) auto_set_seq_no: bool,
+    pub(crate) codecs_from_server: RawSupportedCodecs,
+
+    writer_message_sender: mpsc::Sender<TopicWriterMessage>,
+    writer_loop: JoinHandle<()>,
+    receive_messages_loop: JoinHandle<()>,
+
+    cancellation_token: CancellationToken,
+    writer_state: Arc<Mutex<TopicWriterState>>,
+
+    confirmation_reception_queue: Arc<Mutex<TopicWriterReceptionQueue>>,
+
+    pub(crate) connection_manager: GrpcConnectionManager,
+}
+
+#[allow(dead_code)]
+pub struct AckFuture {}
+
+impl Future for AckFuture {
+    type Output = ();
+
+    fn poll(self: Pin<&mut Self>, _cx: &mut Context<'_>) -> Poll<Self::Output> {
+        unimplemented!("prototype")
+    }
+}
+
+struct WriterPeriodicTaskParams {
+    write_request_messages_chunk_size: usize,
+    write_request_send_messages_period: Duration,
+    auto_set_seq_no: bool,
+    producer_id: Option<String>,
+    request_stream: mpsc::UnboundedSender<stream_write_message::FromClient>,
+}
+
+impl TopicWriter {
+    pub(crate) async fn new(
+        writer_options: TopicWriterOptions,
+        connection_manager: GrpcConnectionManager,
+    ) -> YdbResult<Self> {
+        let mut topic_service = connection_manager
+            .get_auth_service(grpc_wrapper::raw_topic_service::client::RawTopicClient::new)
+            .await?;
+        let init_request_body = InitRequest {
+            path: writer_options.topic_path.clone(),
+            producer_id: writer_options.producer_id.clone().unwrap(), // TODO: handle somehow
+            write_session_meta: writer_options.session_metadata.clone().unwrap_or_default(),
+            get_last_seq_no: writer_options.auto_seq_no.clone(),
+            partitioning: Some(Partitioning::MessageGroupId(
+                writer_options.producer_id.clone().unwrap(),
+            )),
+        };
+
+        let mut stream = topic_service.stream_write(init_request_body).await?;
```

Comment:
Stream initialize must be in background, for use error retry policy. If user need wait of connection for example - wait of last seqno - it must call another func for explicit wait init result.